lib/semian/atomic_integer.rb

```diff
@@ -0,0 +1,26 @@
+module Semian
+  class AtomicInteger < SharedMemoryObject #:nodoc:
```

Comment:
Instead of this `#acquire` trick, how about calling the SysV implementation `SysVAtomicInteger` and in `semian.rb` override the Ruby implementation if semaphores are enabled? This is pretty hard to understand.